app/server/utils/transfer_limits.py

```diff
@@ -157,160 +151,542 @@ def is_any_token_and_user_is_kyc_verified(credit_transfer):
+            regular_payment_filter
+        ).execution_options(show_all=True).first().total or 0
+
+    def throw_validation_error(self, transfer: CreditTransfer, available: AggregateAvailability):
+        message = 'Account Limit "{}" reached. {} available'.format(self.name, max(int(available), 0))
+
+        raise MinimumSentLimitError(
+            transfer_amount_limit=self.available_base(transfer),
+            transfer_amount_avail=available,
+            limit_time_period_days=self.time_period_days,
+            token=transfer.token.name,
+            message=message
+        )
+
+    def __init__(
+            self,
+            name: str,
+            applied_to_transfer_types: AppliedToTypes,
+            application_filter: ApplicationFilter,
+            time_period_days: int,
+            aggregation_filter: AggregationFilter = matching_transfer_type_filter
+    ):
+        super().__init__(
+            name,
+            applied_to_transfer_types,
+            application_filter,
+            time_period_days,
+            aggregation_filter
+        )
+
+
+class BalanceFractionLimit(AggregateTransferAmountMixin, AggregateLimit):
+
+    def available_base(self, transfer: CreditTransfer) -> TransferAmount:
+        return self._total_from_balance(transfer.sender_transfer_account.balance)
+
+    def _total_from_balance(self, balance: int) -> TransferAmount:
+        amount: TransferAmount = int(self.balance_fraction * balance)
+        return amount
+
+    def throw_validation_error(self, transfer: CreditTransfer, available: AggregateAvailability):
+        message = 'Account % Limit "{}" reached. {} available'.format(
+            self.name,
+            max([available, 0])
+        )
+        raise TransferBalanceFractionLimitError(
+            transfer_balance_fraction_limit=self.balance_fraction,
+            transfer_amount_avail=available,
+            limit_time_period_days=self.time_period_days,
+            token=transfer.token.name,
+            message=message
+        )
+
+    def __init__(
+            self,
+            name: str,
+            applied_to_transfer_types: AppliedToTypes,
+            application_filter: ApplicationFilter,
+            time_period_days: int,
+            balance_fraction: float,
+            aggregation_filter: AggregationFilter = matching_transfer_type_filter
+
+    ):
+        super().__init__(
+            name,
+            applied_to_transfer_types,
+            application_filter,
+            time_period_days,
+            aggregation_filter
+        )
+
+        self.balance_fraction = balance_fraction
+
+
+class TransferCountLimit(AggregateLimit):
+
+    def used_aggregator(self, transfer: CreditTransfer, query_constructor: QueryConstructorFunc):
+        return query_constructor(
+            transfer,
+            db.session.query(func.count(CreditTransfer.id).label('count'))
+        ).execution_options(show_all=True).first().count - 1
```

Comment:
Do we need a `show_all` here?

Reply:
My logic is as follows:

TransferAmount aggregation needs it because you shouldn't be able to get around KYC limits just by doing transfers from a different oranisation
There may be cases where a limit for may not want to count across all orgs, but i think it makes more sense to count everything by default, and then filter out with the custom filters if needed.

Reply:
Agree with that logic @enjeyw